in_toto/gpg/util.py

```diff
@@ -100,34 +101,96 @@ def parse_packet_header(data, expected_type=None):
 
     expected_type: (optional)
             Used to error out if the packet does not have the expected
-            type. See in_toto.gpg.constants.PACKET_TYPES for available types.
+            type. See in_toto.gpg.constants.PACKET_TYPE_* for available types.
 
   <Exceptions>
     in_toto.gpg.exceptions.PacketParsingError
+            If the new format packet length encodes a partial body length
+            If the old format packet length encodes an indeterminate length
+            If header or body length could not be determined
             If the expected_type was passed and does not match the packet type
 
+    IndexError
+            If the passed data is incomplete
+
   <Side Effects>
     None.
 
   <Returns>
     The RFC4880-compliant packet payload, its length and its type.
+    (see RFC 4880 4.3. for the list of available packet types)
   """
   data = bytearray(data)
-  packet_type = (data[0] & 0x3c ) >> 2
-  packet_length_bytes = data[0] & 0x03
+  header_len = None
+  body_len = None
+
+  # If Bit 6 of 1st octet is set we parse a New Format Packet Length, and
+  # an Old Format Packet Lengths otherwise
+  if data[0] & 0x40: # pragma: no cover
```

Comment:
Consider using binary representations for this and L132, `0b0100_0000` and `0b0011_1111` and such. (I think this is what they're here for :-)

Reply:
Great idea, will do.